src/fileActions.jsx

```diff
@@ -467,17 +467,25 @@ const downloadFile = (currentPath, selected) => {
     window.open(`${prefix}?${query}`);
 };
 
-export const fileActions = (path, selected, setSelected, clipboard, setClipboard, addAlert, Dialogs) => {
+export const fileActions = (path, selected, setSelected, clipboard, setClipboard, cwdInfo, addAlert, Dialogs) => {
     const currentPath = path.join("/") + "/";
     const menuItems = [];
 
-    const spawnPaste = (sourcePath, targetPath) => {
+    const spawnPaste = (sourcePaths, targetPath) => {
+        if (sourcePaths.some(sourcePath => cwdInfo?.entries[basename(sourcePath)])) {
+            addAlert(_("Paste not overwriting existing files"), "danger", "paste-error");
+            return;
+        }
         cockpit.spawn([
             "cp",
-            "-R",
-            ...sourcePath,
+            "--archive",
```

Comment:
This includes file permissions, which in particular means that it will happily copy suid/sgid binaries or world-writable directories. This also is not desirable.

Reply:
That no longer happens as superuser so is it ok then? Otherwise hmm we need to scrap the whole functionality.

Reply:
Yes, without superuser it seems fine to me.

Reply:
ok!